### PR TITLE
ext/ modules  - use AV _simple functions on new AVs

### PR DESCRIPTION
**Pull request by richardleach**

### CODE REVIEW

**Comment by richardleach** [comment]
I'll make similar changes to more modules under ext/ and add them to this PR. That will hopefully make reviewing more straightforward.

**Comment by demerphq** [comment]
I resolved conflicts on this.

**Comment by jkeenan** [comment]
@richardleach, if you wish to pursue this pull request further, could you resolve merge conflicts and re-push?  Thanks.

File: ext/Hash-Util-FieldHash/FieldHash.xs [hunk 1]
```diff
@@ -159,10 +159,9 @@ static SV *
 HUF_new_trigger(pTHX_ SV *obj, SV *ob_id) {
     dMY_CXT;
     SV* trigger = sv_rvweaken(newRV_inc(SvRV(obj)));
-    AV* cont = newAV();
-    sv_2mortal((SV*)cont);
-    av_store(cont, 0, SvREFCNT_inc(ob_id));
-    av_store(cont, 1, (SV*)newHV());
+    AV* cont = MUTABLE_AV(newSV_type_mortal(SVt_PVAV));
```
Comment (Leont) [inline ext/Hash-Util-FieldHash/FieldHash.xs]: I don't think `MUTABLE_AV(newSV_type_mortal(SVt_PVAV))` is actually helpful for readability. Perhaps we need a `newAV_mortal` macro?

**Comment by jkeenan** [comment]
@richardleach, because this (draft) p.r. touches many different files, it's going to be prone to merge conflicts.  Should you wish to pursue it, could you resolve the current merge conflict(s) and reply to @Leont's question above?  Thanks.

File: ext/Hash-Util-FieldHash/FieldHash.xs (hunk 1 above)
Comment (richardleach) [reply ext/Hash-Util-FieldHash/FieldHash.xs]: Sure thing, I can add one.

**Comment by richardleach** [comment]
> @richardleach, because this (draft) p.r. touches many different files, it's going to be prone to merge conflicts. Should you wish to pursue it, could you resolve the current merge conflict(s) and reply to @Leont's question above? Thanks.

I'll get this done by early in Jan. Thanks.

**Comment by richardleach** [comment]
This has been rebased and Leon's macro suggestion implemented. Does anyone want to give this another once over before it gets merged?

**Comment by iabyn** [comment]
newAV_mortal() needs docs. "make test" currently outputs to stderr:
no docs for newAV_mortal

**Comment by richardleach** [comment]
Thanks for the spot. I'll write something this week.

**Pull request merged**